integration-tests/load-tests/load-tests.spec.ts

```diff
@@ -0,0 +1,153 @@
+import { expect } from 'chai'
+
+/* Imports: External */
+import { Contract, ContractFactory } from 'ethers'
+
+/* Imports: Internal */
+import { OptimismEnv } from '../test/shared/env'
+import {
+  executeRepeatedL1ToL2Transactions,
+  executeRepeatedL2ToL1Transactions,
+  executeRepeatedL2Transactions,
+  executeRepeatedL1ToL2TransactionsParallel,
+  executeRepeatedL2ToL1TransactionsParallel,
+  executeRepeatedL2TransactionsParallel,
+} from './load-test-helpers'
+
+/* Imports: Artifacts */
+import l1SimpleStorageJson from '../artifacts/contracts/SimpleStorage.sol/SimpleStorage.json'
+import l2SimpleStorageJson from '../artifacts-ovm/contracts/SimpleStorage.sol/SimpleStorage.json'
+
+describe('load tests', () => {
+  let env: OptimismEnv
+  before(async () => {
+    env = await OptimismEnv.new()
+  })
+
+  let L2SimpleStorage: Contract
+  let L1SimpleStorage: Contract
+  beforeEach(async () => {
+    const factory__L1SimpleStorage = new ContractFactory(
+      l1SimpleStorageJson.abi,
+      l1SimpleStorageJson.bytecode,
+      env.l1Wallet
+    )
+    const factory__L2SimpleStorage = new ContractFactory(
+      l2SimpleStorageJson.abi,
+      l2SimpleStorageJson.bytecode,
+      env.l2Wallet
+    )
+    L1SimpleStorage = await factory__L1SimpleStorage.deploy()
+    await L1SimpleStorage.deployTransaction.wait()
+    L2SimpleStorage = await factory__L2SimpleStorage.deploy()
+    await L2SimpleStorage.deployTransaction.wait()
+  })
+
+  describe('L1 => L2 load tests', () => {
+    const numTransactions = 10
+
+    it(`${numTransactions} L1 => L2 transactions (serial)`, async () => {
+      await executeRepeatedL1ToL2Transactions(
+        env,
+        {
+          contract: L2SimpleStorage,
+          functionName: 'setValue',
+          functionParams: [`0x${'42'.repeat(32)}`],
+        },
+        numTransactions
+      )
+
+      expect((await L2SimpleStorage.totalCount()).toNumber()).to.equal(
+        numTransactions
+      )
+    }).timeout(400000)
+
+    it.only(`${numTransactions} L1 => L2 transactions (parallel)`, async () => {
+      await executeRepeatedL1ToL2TransactionsParallel(
+        env,
+        {
+          contract: L2SimpleStorage,
+          functionName: 'setValue',
+          functionParams: [`0x${'42'.repeat(32)}`],
+        },
+        numTransactions
+      )
+
+      expect((await L2SimpleStorage.totalCount()).toNumber()).to.equal(
+        numTransactions
+      )
+    }).timeout(400000)
+  })
+
+  describe('L2 => L1 load tests', () => {
+    const numTransactions = 100
```

Comment:
I'd be conscious of what these numbers end up being — if we can get it to be as low as possible while still demonstrating concurrency (e.g., maybe 5?) it would be something we can confidently add to our CI per commit.

Reply:
I set this to 10 for now. We probably want to make it configurable (?) so that we can execute these tests against live deployments with a much higher load.